File: PYTHON/opentutorials/ObjectOrientedProgramming.py
class Cal(object):
    def __init__(self, v1, v2):
        self.v1=v1
        self.v2=v2
    def add(self):
        return self.v1+self.v2
    def sub(self):
        return self.v1-self.v2
    def setValue(self, v):
        if isinstance(v, int):
            self.v1=v
    def getValue(self):
        return self.v1


c1=Cal(10, 10)
print(c1.add())
print(c1.sub())
c1.setValue('one')
#c1.v1='one'
c1.v2=30
print(c1.add())

File: RUBY/opentutorials/ObjectOrientedProgramming.rb
class Cal

  attr_reader :v1
  attr_writer :v1
  attr_accessor :v1

  def initialize(v1, v2)
    @v1=v1
    @v2=v2
  end
  def add()
    return @v1+@v2
  end
  def sub()
    return @v1-@v2
  end
  def setValue(v1)
    if v1.is_a?(Integer)
      @value=v1
    end
  end
  def getValue()
    return @v1
  end
end


c1=Cal.new(10, 10)
p c1.add()
p c1.sub()
c1.setValue('one')
p c1.add()
p c1.getValue()
